### ISSUE TITLE
DXE-4054 Provider producing an inconsistent final plan with Cloudlet policy

### ISSUE THREAD

**Issue by antoine-savignac-nm** [akamai/terraform-provider-akamai#567]
The provider is producing an inconsistent final plan when trying to update a cloudlet policy at the same time as updating a property.
For example we're updating our Edge redirector policy by adding a new rule, and we're also updating our property by modifying an existing rule. When this happens, the cloudlet policy is updated first and then terraform is crashing with the following error message: `"registry.terraform.io/akamai/akamai" produced an invalid new value for .production_version: was known but now unknown.`

### Terraform and Akamai Terraform Provider Versions
Terraform version: 1.2.5
Akamai provider version: 6.2.0

### Affected Resource(s)
- akamai_cloudlets_policy
- akamai_cloudlets_policy_activation
- akamai_property

### Terraform Configuration Files
```hcl
resource "akamai_property" "property" {
  name        = "name"
  contract_id = data.akamai_contract.contract.id
  group_id    = data.akamai_group.group.id
  product_id  = "product_id"

  rule_format = "v2024-01-09"
  rules       = data.akamai_property_rules_builder.rules.json
}

resource "akamai_cloudlets_policy" "redirections" {
  name          = "redirections"
  cloudlet_code = "ER"
  description   = ""
  group_id      = "groupId"
  match_rules   = data.akamai_cloudlets_edge_redirector_match_rule.redirections.json
  is_shared     = true
}

resource "akamai_cloudlets_policy_activation" "redirections" {
  policy_id = akamai_cloudlets_policy.redirections_nutmeg_gbr_bigbear.id
  network   = "production"
  version   = akamai_cloudlets_policy.redirections.version
}

data "akamai_property_rules_builder" "redirections_nutmeg_gbr_bigbear" {
  rules_v2024_01_09 {
    name                  = "Edge Redirector"
    criteria_must_satisfy = "all"
    behavior {
      edge_redirector {
        enabled                = true
        is_shared_policy       = true
        cloudlet_shared_policy = akamai_cloudlets_policy.redirections.id
      }
    }
  }
}
```

### Panic Output
```
11:22:41 │ Error: Provider produced inconsistent final plan
11:22:41 │ 
11:22:41 │ When expanding the plan for
11:22:41 │ module.property.akamai_property.property to
11:22:41 │ include new values learned so far during apply, provider
11:22:41 │ "registry.terraform.io/akamai/akamai" produced an invalid new value for
11:22:41 │ .production_version: was known, but now unknown.
11:22:41 │ 
11:22:41 │ This is a bug in the provider, which should be reported in the provider's
11:22:41 │ own issue tracker.
11:22:41 ╵
11:22:41 ╷
11:22:41 │ Error: Provider produced inconsistent final plan
11:22:41 │ 
11:22:41 │ When expanding the plan for
11:22:41 │ module.property.akamai_property.property to
11:22:41 │ include new values learned so far during apply, provider
11:22:41 │ "registry.terraform.io/akamai/akamai" produced an invalid new value for
11:22:41 │ .staging_version: was known, but now unknown.
11:22:41 │ 
11:22:41 │ This is a bug in the provider, which should be reported in the provider's
11:22:41 │ own issue tracker.
11:22:41 ╵
11:22:41 ╷
11:22:41 │ Error: Provider produced inconsistent final plan
11:22:41 │ 
11:22:41 │ When expanding the plan for
11:22:41 │ module.property.akamai_property.property to
11:22:41 │ include new values learned so far during apply, provider
11:22:41 │ "registry.terraform.io/akamai/akamai" produced an invalid new value for
11:22:41 │ .latest_version: was known, but now unknown.
11:22:41 │ 
11:22:41 │ This is a bug in the provider, which should be reported in the provider's
11:22:41 │ own issue tracker.
```

### Expected Behavior
The cloudlet policy should have been updated properly, then the property should also have been updated

### Actual Behavior
The cloudlet policy was updated but then the terraform job crashed

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
This only happens when updating BOTH the property and the cloudlet policy at the same time, when we're updating only 1 of those components then it is fine.
Running `terraform apply` again after the failure fixes the issue.

**Comment by KonradRajda** [comment]
Hi @antoine-savignac-nm 
Thanks for reporting the issue. We were able to reproduce it and it will be investigated. We will come back to you.

**Comment by lkowalsk-akamai-com** [comment]
Hi @antoine-savignac-nm , this should be fixed in the recent 6.4.0 release.
